### PR TITLE
[SPARK-36259] Expose localtimestamp in pyspark.sql.functions

### PR DESCRIPTION
**Pull request by dominikgehl**
### What changes were proposed in this pull request?
Exposing localtimestamp in pyspark.sql.functions

### Why are the changes needed?
Was previously only available in scala

### Does this PR introduce _any_ user-facing change?
new localtimestamp in pyspark.sql.functions

### How was this patch tested?
test added inline

### CODE REVIEW

**Comment by MaxGekk** [comment]
So far, we shouldn't expose the function, see https://github.com/apache/spark/pull/33444 . cc @gengliangwang

File: python/pyspark/sql/types.py [hunk 1]
```diff
@@ -899,6 +899,8 @@ def _parse_datatype_json_value(json_value):
             return _all_atomic_types[json_value]()
         elif json_value == 'decimal':
             return DecimalType()
+        elif json_value == 'timestamp_ntz':
+            return TimestampType()
```
Comment (gengliangwang) [inline python/pyspark/sql/types.py:903]: I think we need a design for how to support TimestampNTZ type in PySpark. Here the `timestamp_ntz` is mapped to `TimestampType()`
cc @HyukjinKwon

Comment (HyukjinKwon) [reply python/pyspark/sql/types.py:903]: Yeah, let's add the proper support of `TimestampTypeNTZ` first.

File: python/pyspark/sql/types.py [hunk 2]
```diff
@@ -186,6 +187,20 @@ def fromInternal(self, ts):
             return datetime.datetime.fromtimestamp(ts // 1000000).replace(microsecond=ts % 1000000)
 
 
+class TimestampNTZType(TimestampType):
```
Comment (HyukjinKwon) [inline python/pyspark/sql/types.py:190]: @dominikgehl can we do this in a separate PR? With adding this type, we should also check if it works w/:
- `SparkSession.createDataFrame` (when 'spark.sql.execution.arrow.pyspark.enabled'` configuration is on and off)
- `DataFrame.toPandas` (when 'spark.sql.execution.arrow.pyspark.enabled'` configuration is on and off)
- `DataFrame.collect()`
- `pyspark.functions.udf`
- `pyspark.functions.pandas_udf`

File: python/pyspark/sql/types.py (hunk 1 above)
Comment (dominikgehl) [reply python/pyspark/sql/types.py]: Updated with new TimestampNTZType

File: python/pyspark/sql/types.py (hunk 2 above)
Comment (dominikgehl) [reply python/pyspark/sql/types.py:190]: @HyukjinKwon Just to get the workflow right, should I create a separate JIRA ticket to create the TimestampNTZType ? And then once we have the additional type, we come back to this PR ?

Comment (HyukjinKwon) [reply python/pyspark/sql/types.py:190]: Yes, let;s create a new ticket for `TimestampNTZType`. Yes, let's do that.

Comment (dominikgehl) [reply python/pyspark/sql/types.py:190]: New ticket is https://issues.apache.org/jira/projects/SPARK/issues/SPARK-36287

**Comment by HyukjinKwon** [comment]
Let's implement https://issues.apache.org/jira/projects/SPARK/issues/SPARK-36287 first before getting this in.

**Pull request closed without merging**